Sources/Starknet/Network/HttpNetworkProvider.swift

```diff
@@ -47,16 +47,30 @@ class HttpNetworkProvider {
         return request
     }
 
+    private func handleEncodingError(_ error: Error) throws -> Never {
```

Comment:
As [official documentation](https://developer.apple.com/documentation/swift/never) explains:

> Use Never as the return type when declaring a closure, function, or method that unconditionally throws an error, traps, or otherwise does not terminate.

Without marking `Never` here, the code would look like that:
```swift
try handleEncoding(...)
fatalError(...)
``` 

Imo using `Never` is more suitable in this case. Wdyt? @DelevoXDG